### PR TITLE
Docs: Update PR creation bot message

### PR DESCRIPTION
**Pull request by nzakas**
This updates our pull request message to do some basic checks:

1. Commit message format (tag, issue reference, length)
1. CLA signature
1. Number of commits

If there are no problems, the comment is just "LGTM". :)

Right now, it can't check commit messages because the bot needs to do a separate request to get that information. I'll update the bot to do so once this is merged.

Also, I'll update the bot to run this on every pull request, not just on those without a CLA (as it is now).

### CODE REVIEW

File: templates/pr-create.md.ejs [hunk 1]
```diff
@@ -1,5 +1,55 @@
...
+
+var problems = [];
+
+// Check for one commit per pull request
+if (payload.commits > 1) {
+    problems.push("We require one commit per pull request. Please [squash](https://egghead.io/lessons/javascript-how-to-squash-multiple-git-commits) your commits.");
+} else if (meta.commits) {
+    var log = meta.commits[0].commit.message;
+    if (!isValidCommitFlag(log)) {
+        problems.push("The commit summary needs to begin with a tag (such as `Fix:` or `Update:`). Please check out our [guide](http://eslint.org/docs/developer-guide/contributing/pull-requests#step-2-make-your-changes) for how to properly format your commit summary.")
```
Comment (ilyavolodin) [inline templates/pr-create.md.ejs]: Can we add some quick instruction on how to do amend to pull requests? That information is not really all that easy to find, and I think would be helpful. Or at least a link to http://eslint.org/docs/developer-guide/contributing/pull-requests#updating-the-commit-message

Comment (nzakas) [inline templates/pr-create.md.ejs]: Good point, I'll add that.

**Comment by ilyavolodin** [comment]
LGTM

File: templates/pr-create.md.ejs [hunk 2]
```diff
@@ -1,5 +1,53 @@
-Thanks for the pull request, @<%= payload.sender.login %>! It looks like this is your first time contributing to ESLint, so please take a moment to read over our [contribution guidelines](http://eslint.org/docs/developer-guide/contributing/pull-requests).
+<%
 
-We'll also need you to sign our [CLA](http://eslint.org/cla), which is just a way for you to say that you give us permission to use your contribution.
+var FLAG_PATTERN = /^(Fix|Update|New|Breaking|Build|Docs|Upgrade):/,
+    ISSUE_REF_PATTERN = /\((fixes|refs) #\d+.*?$/;
```
Comment (platinumazure) [inline templates/pr-create.md.ejs]: Shouldn't this have a `\)`? (I know it's covered by `.*?` but if we're mandating open parentheses, we should probably check for closed parentheses too)

Comment (nzakas) [inline templates/pr-create.md.ejs]: Yeah, good catch. I'll update.

File: templates/pr-create.md.ejs [hunk 3]
```diff
@@ -1,5 +1,53 @@
-Thanks for the pull request, @<%= payload.sender.login %>! It looks like this is your first time contributing to ESLint, so please take a moment to read over our [contribution guidelines](http://eslint.org/docs/developer-guide/contributing/pull-requests).
+<%
 
-We'll also need you to sign our [CLA](http://eslint.org/cla), which is just a way for you to say that you give us permission to use your contribution.
+var FLAG_PATTERN = /^(Fix|Update|New|Breaking|Build|Docs|Upgrade):/,
```
Comment (BYK) [inline templates/pr-create.md.ejs]: I wish we could share some logic between here and `npm run check-commit` :(

Comment (BYK) [inline templates/pr-create.md.ejs]: Also, right now the build would still fail if only the commit message is not proper right?

Comment (ilyavolodin) [inline templates/pr-create.md.ejs]: Correct

**Pull request merged**